File: examples/typescript/guides/integrations/embedding-cohere.ts
// Third-party integration - requires: pnpm add cohere-ai
// See: /guides/rag-and-indexing

import Cohere from 'cohere-ai';

const client = new Cohere();

// > Cohere embedding usage
export async function embed(text: string): Promise<number[]> {
  const r = await client.embed({
    texts: [text],
    model: 'embed-english-v3.0',
    inputType: 'search_document',
  });
  return r.embeddings[0] ?? [];
}

File: examples/python/guides/integrations/embedding_cohere.py
# Third-party integration example - requires: pip install cohere
# See: /guides/rag-and-indexing

import cohere

client = cohere.Client()


# > Cohere embedding usage
def embed(text: str) -> list[float]:
    r = client.embed(texts=[text], model="embed-english-v3.0", input_type="search_document")
    return list(r.embeddings[0])
